backend/src/impl/default_controllers_impl.py

```diff
@@ -219,8 +231,24 @@ def benchmark_get_by_id(benchmark_id: str, by_creator: bool) -> Benchmark:
 """ /systems """
 
 
-def systems_get_by_id(system_id: str) -> System:
-    return SystemDBUtils.find_system_by_id(system_id)
+def systems_get_by_id(system_id: str, with_user_info: bool = None) -> System:
+    system = SystemDBUtils.find_system_by_id(system_id)
+    user = get_user()
+    # must be creator to see user info
+    if with_user_info:
+        has_access = user.is_authenticated and _is_creator(system, user)
+        if not has_access:
+            abort_with_error_message(403, "system access denied", 40302)
+        else:
+            return system
+    else:
+        has_access = user.is_authenticated and (
+            _is_creator(system, user) or _is_shared_user(system, user)
```

Comment:
As this `or` statement is used widely, maybe it could be replaced by a single function call `_has_read_access(system, user)`